### PR TITLE
Add a badge option for tabs

### PR DESCRIPTION
**Pull request by timshadel**

### CODE REVIEW

File: CustomTabBar/TabButton.swift [hunk 1]
```diff
@@ -156,7 +164,21 @@ private extension TabButton {
             titleLabel.font = UIFont.systemFont(ofSize: 16)
         }
     }
-    
+
+    func updateBadge() {
+        guard let value = dataObject.badgeValue else { badgeLabel.removeFromSuperview(); return }
+        badgeLabel.text = value
+        badgeLabel.textAlignment = .center
+        badgeLabel.backgroundColor = .red
```
Comment (benjaminsnorris) [inline CustomTabBar/TabButton.swift]: I think we'd want to parameterize this. Different designs will probably call for slightly different shades of red with their color scheme.

File: CustomTabBar/TabButton.swift [hunk 2]
```diff
@@ -156,7 +164,21 @@ private extension TabButton {
...
+        if let badgeFont = badgeFont {
+            badgeLabel.font = badgeFont
+        } else {
+            badgeLabel.font = UIFont.boldSystemFont(ofSize: 12)
+        }
+        badgeLabel.textColor = .white
```
Comment (benjaminsnorris) [inline CustomTabBar/TabButton.swift]: It'd be good to have this as a parameter too. Assuming that we have the text color of the tabs as a parameter. If not, let's not worry about this either.

File: CustomTabBar/TabButton.swift [hunk 3]
```diff
@@ -165,6 +187,24 @@ private extension TabButton {
...
+        guard badgeLabel.superview == nil else { return }
+        addSubview(badgeLabel)
+        badgeLabel.translatesAutoresizingMaskIntoConstraints = false
+        if let imageAnchor = imageButton.imageView?.trailingAnchor {
+            badgeLabel.centerXAnchor.constraint(equalTo: imageAnchor, constant: TabButton.badgeMargin).isActive = true
+        } else {
+            badgeLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -TabButton.badgeMargin).isActive = true
+        }
+        badgeLabel.topAnchor.constraint(equalTo: topAnchor, constant: TabButton.badgeMargin).isActive = true
+        let badgeHeight = badgeLabel.intrinsicContentSize.height * 1.1
```
Comment (benjaminsnorris) [inline CustomTabBar/TabButton.swift:211]: Do we need to do a layout pass before this has useful information? Or does this already get the right thing by setting the content?

**Review by benjaminsnorris — CHANGES_REQUESTED** [review]
This is really exciting. Let's parameterize the design a bit more with the colors, and then I think it's good!

File: CustomTabBar/TabButton.swift (hunk 3 above)
Comment (timshadel) [reply CustomTabBar/TabButton.swift:211]: I dunno. It seems like it worked all the time for me. I _assume_ that the intrinsic size doesn't change after the width is set, but I dunno for sure.

**Review by benjaminsnorris — APPROVED** [review]

**Pull request merged**